pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -0,0 +1,339 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import importlib
+import io
+import math
+import os
+import sys
+from typing import Optional, Union
+
+# check if ipywidgets is installed before importing tqdm.auto
+# to ensure it won't fail and a progress bar is displayed
+if importlib.util.find_spec("ipywidgets") is not None:
+    from tqdm.auto import tqdm as _tqdm
+else:
+    from tqdm import tqdm as _tqdm
+
+from pytorch_lightning.callbacks.progress.base import ProgressBarBase
+
+_PAD_SIZE = 5
+
+
+class tqdm(_tqdm):
```

Comment:
Class shall start with CAPITALS

Reply:
I think it's fine here as we are trying to transparently override tqdm, not sure it makes sense to change the name.

Reply:
yes, but it make confusion as you have identical class name for two different classes

Reply:
in any case, it can't be changed in this PR

Reply:
Yup, closing this thread. As the suggested change is not related to the PR.